### PR TITLE
Add custom dimension for authenticated or not users.

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue:

- #4846 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: includes/Core/Authentication/Authentication.php [hunk 1]
```diff
@@ -364,11 +364,11 @@ function( $user ) {
 					$user['user']['email']   = $profile_data['email'];
 					$user['user']['picture'] = $profile_data['photo'];
 				}
-
-				$user['connectURL']     = esc_url_raw( $this->get_connect_url() );
-				$user['initialVersion'] = $this->initial_version->get();
-				$user['userInputState'] = $this->user_input_state->get();
-				$user['verified']       = $this->verification->has();
+				$user['user']['authenticated'] = $this->is_authenticated();
```
Comment (aaemnnosttv) [inline includes/Core/Authentication/Authentication.php:367]: The `user` key/property is for data specific to the WP User (with the exception of the email and picture 😄 ). Let's put this on the top-level, as a sibling of `verified` instead.

File: assets/js/util/tracking/createTrackEvent.js [hunk 2]
```diff
@@ -64,6 +64,10 @@ export default function createTrackEvent(
 			dimension4: global.GOOGLESITEKIT_VERSION || '',
 			dimension5: Array.from( enabledFeatures ).join( ',' ),
 			dimension6: activeModules.join( ',' ),
+			// Custom dimension to differentiate between authenticated and non authenticated users.
+			dimension7: global._googlesitekitUserData?.user?.authenticated
```
Comment (aaemnnosttv) [inline assets/js/util/tracking/createTrackEvent.js:68]: I realize this was done once before but we should avoid reaching for globals directly here. [We use a number of globals already](https://github.com/google/site-kit-wp/blob/460a21e240c91f51f6a22ca17b1ba67b8162236e/assets/js/util/tracking/index.js#L24-L31) but these are passed through in the `config`, so let's take the same approach here.

Would you please do the same with the `currentUserRoles` above for consistency?

We still reference `console` and `_gaUserPrefs` on the global below but even then we use the injected `_global`, but these are the only exceptions we should need.

Also – please add/update the tests for `trackEvent` to ensure this dimension is covered.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @asvinb – this one is mostly there, just a few small things to address 👍

File: assets/js/util/tracking/createTrackEvent.js (hunk 2 above)
Comment (asvinb) [reply assets/js/util/tracking/createTrackEvent.js]: I've also updated the use of `GOOGLESITEKIT_VERSION` as well.

File: includes/Core/Authentication/Authentication.php (hunk 1 above)
Comment (asvinb) [reply includes/Core/Authentication/Authentication.php]: 👍

File: assets/js/util/tracking/index.js [hunk 3]
```diff
@@ -30,13 +30,23 @@ const {
...
+const { GOOGLESITEKIT_VERSION: pluginVersion } = global;
+
 const initialConfig = {
 	activeModules,
 	trackingEnabled,
 	trackingID,
 	referenceSiteURL,
 	userIDHash,
 	isSiteKitScreen,
+	currentUserRoles,
```
Comment (aaemnnosttv) [inline assets/js/util/tracking/index.js:47]: We can leave off the `current` as everything related to the user here is already specific to the current user (e.g. `userIDHash`)
```suggestion
	userRoles,
```

File: assets/js/util/tracking/index.js [hunk 4]
```diff
@@ -30,13 +30,23 @@ const {
...
+	authenticatedUser,
```
Comment (aaemnnosttv) [inline assets/js/util/tracking/index.js:48]: Let's use `isAuthenticated` for consistency with other uses/references in JS.
```suggestion
	isAuthenticated,
```

File: assets/js/util/tracking/createTrackEvent.js [hunk 5]
```diff
@@ -61,9 +61,11 @@ export default function createTrackEvent(
 			dimension1: referenceSiteURL,
 			dimension2: currentUserRoles.join( ',' ),
 			dimension3: userIDHash,
-			dimension4: global.GOOGLESITEKIT_VERSION || '',
+			dimension4: pluginVersion || '',
 			dimension5: Array.from( enabledFeatures ).join( ',' ),
 			dimension6: activeModules.join( ',' ),
+			// Custom dimension to differentiate between authenticated and non authenticated users.
```
Comment (aaemnnosttv) [inline assets/js/util/tracking/createTrackEvent.js:67]: I know the IB mentions this but I'm not sure the comment adds much value as it reads to be rather self-documenting IMO 😄

File: assets/js/util/tracking/index.test.js [hunk 6]
```diff
@@ -85,6 +85,9 @@ describe( 'trackEvent', () => {
 			trackingID: 'UA-12345678-1',
 			activeModules: [],
 			trackingEnabled: true,
+			currentUserRoles: [ 'administrator' ],
+			authenticatedUser: 1,
+			pluginVersion: global.GOOGLESITEKIT_VERSION,
```
Comment (aaemnnosttv) [inline assets/js/util/tracking/index.test.js:90]: Let's use an inline value here for the test rather than use the global.
```suggestion
			pluginVersion: '1.2.3',
```

File: includes/Core/Assets/Assets.php [hunk 7]
```diff
@@ -452,6 +452,7 @@ private function get_assets() {
 					'src'          => $base_url . 'js/googlesitekit-base.js',
 					'dependencies' => array(
 						'googlesitekit-base-data',
+						'googlesitekit-user-data',
```
Comment (aaemnnosttv) [inline includes/Core/Assets/Assets.php:455]: Why was this added? The base JS doesn't use `trackEvent` or reference the user global.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @asvinb – just a few minor things left to address 👍

File: assets/js/util/tracking/index.js (hunk 3 above)
Comment (asvinb) [reply assets/js/util/tracking/index.js]: 👍

File: assets/js/util/tracking/index.js (hunk 4 above)
Comment (asvinb) [reply assets/js/util/tracking/index.js]: 👍

File: assets/js/util/tracking/createTrackEvent.js (hunk 5 above)
Comment (asvinb) [reply assets/js/util/tracking/createTrackEvent.js]: 😁 Removed!

File: includes/Core/Assets/Assets.php (hunk 7 above)
Comment (asvinb) [reply includes/Core/Assets/Assets.php:455]: @aaemnnosttv It's because upon page load, `window._googlesitekitUserData` is `undefined`

Comment (aaemnnosttv) [reply includes/Core/Assets/Assets.php:455]: I see what you mean. I wonder if these properties belong in base data instead then? 🤔 We probably should avoid loading all user data on all of wp-admin.

@felixarntz do you think we should keep all the metadata for custom dimensions in the base data or load an extra data dependency?

Comment (felixarntz) [reply includes/Core/Assets/Assets.php:455]: @aaemnnosttv @asvinb Let's put it in `_googlesitekitBaseData`, like all other tracking dependency data. `userIDHash` for example is also technically user information, but part of the base data. That data isn't sensitive and should preferably be exlusively used for the tracking, so I think that's preferable over loading the overall user data everywhere.

Comment (aaemnnosttv) [reply includes/Core/Assets/Assets.php:455]: Thanks @felixarntz that sounds good to me 👍 

@asvinb please move `userRoles` to the base data as well so that tracking configuration only depends on the base data for now. In the future, we'll update it to use a separate global just for tracking data.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @asvinb – this is almost there, just the one change left to make 😄 See Felix's comment here https://github.com/google/site-kit-wp/pull/5089/files#r854463054

File: includes/Core/Assets/Assets.php (hunk 7 above)
Comment (asvinb) [reply includes/Core/Assets/Assets.php]: Sounds good @aaemnnosttv . I've updated the PR. Can you take another look?

**Comment by aaemnnosttv** [comment]
Thanks @asvinb this LGTM, just a few minor places where a few of the moved values were still referenced. I'll push these last small tweaks and then this should be good to go 👍

**Review by aaemnnosttv — APPROVED** [review]
